java/client/src/main/java/glide/api/models/commands/ZaddOptions.java

```diff
@@ -0,0 +1,91 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.SortedSetBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Optional arguments to {@link SortedSetBaseCommands#zadd(String, Map, ZaddOptions, boolean)},
+ * {@link SortedSetBaseCommands#zadd(String, Map, ZaddOptions)} and {@link
+ * SortedSetBaseCommands#zaddIncr(String, String, double, ZaddOptions)}
+ *
+ * @see <a href="https://redis.io/commands/zadd/">redis.io</a>
+ */
+@Builder
+public final class ZaddOptions {
+    /**
+     * Defines conditions for updating or adding elements with {@link SortedSetBaseCommands#zadd}
+     * command.
+     */
+    private final ConditionalChange conditionalChange;
+
+    /**
+     * Specifies conditions for updating scores with zadd {@link SortedSetBaseCommands#zadd} command.
+     */
+    private final UpdateOptions updateOptions;
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum ConditionalChange {
+        /**
+         * Only update elements that already exist. Don't add new elements. Equivalent to <code>XX
+         * </code> in the Redis API.
+         */
+        ONLY_IF_EXISTS("XX"),
+        /**
+         * Only add new elements. Don't update already existing elements. Equivalent to <code>NX</code>
+         * in the Redis API.
+         */
+        ONLY_IF_DOES_NOT_EXIST("NX");
+
+        private final String redisApi;
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum UpdateOptions {
+        /**
+         * Only update existing elements if the new score is less than the current score. Equivalent to
+         * <code>LT</code> in the Redis API.
+         */
+        SCORE_LESS_THAN_CURRENT("LT"),
+        /**
+         * Only update existing elements if the new score is greater than the current score. Equivalent
+         * to <code>GT</code> in the Redis API.
+         */
+        SCORE_GREATER_THAN_CURRENT("GT");
+
+        private final String redisApi;
+    }
+
+    /**
+     * Converts ZaddOptions into a String[].
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        if (conditionalChange == ConditionalChange.ONLY_IF_DOES_NOT_EXIST && updateOptions != null) {
+            throw new IllegalArgumentException(
+                    "The GT, LT, and NX options are mutually exclusive. Cannot choose both "
+                            + updateOptions.redisApi
+                            + " and NX.");
```

Comment:
You need tests which cover all 4 combinations. You can update client tests by using different combinations there + add UT for this class (for coverage).

Reply:
Add a UT which covers this case. It is good to create another UT class for zaddoptions only.

Reply:
I did add a test for it in RedisClientTest

Reply:
https://github.com/Bit-Quill/glide-for-redis/pull/115/files#diff-976ae060a9ceecb773b36ba8fa33376c9bb012f2697e105aceda55a25b191b4bR914